### PR TITLE
SKA-13: Created endpoint and service for getting hourly cloud data

### PR DESCRIPTION
**Pull request by rohan-sahgal**
# Description

Created new endpoint for getting hourly cloud data from OpenWeather, as well as some minor changes to other methods in the service.

## Type of change

- [x] New feature (non-breaking change which adds functionality).
- [x] This change requires a documentation update.

# How Has This Been Tested?

Tested locally by running the service and testing the endpoint in Swagger.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: sky_alert/endpoint.py [hunk 1]
```diff
@@ -18,11 +19,22 @@ def healthz() -> str:
...
+@app.get("/openweather_cloud_data")  # type: ignore
+def cloud_data(lat: str = "0", lon: str = "0") -> Union[CloudData, int]:
+    return ows.get_cloud_data(lat=lat, lon=lon)
+
+
+@app.get("/get_and_write_openweather_data_to_file")  # type: ignore
+def get_and_write_openweather_data_to_file(lat: str = "0", lon: str = "0") -> None:
+    ows.update_most_recent_weather(lat=lat, lon=lon)
+    with open("most_recent_weather.json", "w") as f:
+        json.dump(ows.most_recent_weather[(lat, lon)], f)
```
Comment (thereisnoaddress) [inline sky_alert/endpoint.py:40]: Maybe we could add something to indicate whether writing failed, ie 
```suggestion
    ows.update_most_recent_weather(lat=lat, lon=lon)
    with open("most_recent_weather.json", "w") as f:
        json.dump(ows.most_recent_weather[(lat, lon)], f)
    print("writing succeeded")
    return True
```

Comment (thereisnoaddress) [reply sky_alert/endpoint.py:40]: something better worded

File: sky_alert/locations.txt
Comment (thereisnoaddress) [inline sky_alert/locations.txt:1]: should we consider refactoring this into a `constants.py` file? 

ie 

```
COORDINATES_NIAGARA_ON_THE_LAKE=(43.249320405808895, -79.08154178737297)
...
```

and when we need to use it, we can do 
```
from sky_alert.constants import COORDINATES_NIAGARA_ON_THE_LAKE
...
```

File: sky_alert/openweather_service.py [hunk 2]
```diff
@@ -97,6 +97,31 @@ def get_moonrise_moonset_from_json(self, lat: str, lon: str) -> MoonData:
...
+        cloud_data = []
+
+        self.update_most_recent_weather(lat=lat, lon=lon)
+
+        json_data = self.most_recent_weather[(lat, lon)]
+
+        # Check for key existence at different levels
+        if "hourly" in json_data and isinstance(json_data["hourly"], list):
+            current_data = json_data["hourly"]
+            for i in range(0, 24):
```
Comment (thereisnoaddress) [inline sky_alert/openweather_service.py:112]: nit, but we could declare something like `HOURS_IN_DAY=24` ~~on top of the file~~ in the `constants.py` file and then do 
```suggestion
            for i in range(HOURS_IN_DAY):
```

File: sky_alert/endpoint.py [hunk 3]
```diff
@@ -18,11 +19,22 @@ def healthz() -> str:
...
 
 @app.get("/openweather_moon_data")  # type: ignore
 def moon_data(lat: str = "0", lon: str = "0") -> Union[MoonData, int]:
-    return ows.get_moonrise_moonset_from_json(lat=lat, lon=lon)
+    return ows.get_moon_data(lat=lat, lon=lon)
+
+
+@app.get("/openweather_cloud_data")  # type: ignore
+def cloud_data(lat: str = "0", lon: str = "0") -> Union[CloudData, int]:
+    return ows.get_cloud_data(lat=lat, lon=lon)
```
Comment (thereisnoaddress) [inline sky_alert/endpoint.py:33]: clean refactor! 🧹

**Review by thereisnoaddress — COMMENTED** [review]
clean feature add + refactoring!

a few comments, mostly related to the constants

**Comment by rohan-sahgal** [comment]
Skipped mypy due to following error message:

```
(.venv) rohan@DESKTOP-65QNOCU:~/sky-alert$ git commit -m "Made changes as per Chris' suggestions"
black....................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

pyproject.toml:1: error: Error importing plugin "pydantic.mypy": No module named 'pydantic'  [misc]
Found 1 error in 1 file (errors prevented further checking)
```

**Review by thereisnoaddress — APPROVED** [review]

**Pull request merged**